crates/worker/src/commands/utils.rs

```diff
@@ -0,0 +1,183 @@
+) -> Result<(), PreparationError> {
+    let db_update_result = db_client
+        .client
+        .update_item()
+        .table_name(db_client.table_name.clone())
+        .key(Item::primary_key_name(), AttributeValue::S(key.to_string()))
+        .update_expression("SET #status = :newStatus")
+        .condition_expression("#status = :currentStatus")
+        .expression_attribute_names("#status", Status::attribute_name())
+        .expression_attribute_values(
+            ":newStatus",
+            AttributeValue::N(u32::from(Status::InProgress).to_string()),
+        )
+        .expression_attribute_values(
+            ":currentStatus",
+            AttributeValue::N(u32::from(Status::Pending).to_string()),
+        )
+        .send()
+        .await;
+
+    match db_update_result {
+        Ok(_) => Ok(()),
+        Err(SdkError::ServiceError(err)) => match err.err() {
+            UpdateItemError::ConditionalCheckFailedException(_) => {
+                error!("Conditional check not met");
+                Err(PreparationError::UnexpectedStatusError(
+                    "Concurrent status change from another instance".into(),
+                ))
+            }
+            _ => Err(DBError::from(SdkError::ServiceError(err)).into()),
+        },
+        Err(err) => Err(DBError::from(err).into()),
+    }
+}
+
+pub(crate) async fn prepare_compile_input(
+    request: &CompilationRequest,
+    db_client: &DynamoDBClient,
+    s3_client: &S3Client,
+) -> Result<CompilationInput, PreparationError> {
+    let zksolc_version = request.config.version.as_str();
+    if !ZKSOLC_VERSIONS.contains(&zksolc_version) {
+        return Err(PreparationError::VersionNotSupported(
+            zksolc_version.to_string(),
+        ));
+    }
+
+    let item = db_client.get_item(request.id.to_string().as_str()).await?;
+    let item: Item = match item {
+        Some(item) => item,
+        None => {
+            error!("No item id: {}", request.id);
+            return Err(PreparationError::NoDBItemError(request.id.to_string()));
+        }
+    };
+
+    match item.status {
+        Status::Pending => {}
+        status => {
+            warn!("Item already processing: {}", status);
+            return Err(PreparationError::UnexpectedStatusError(status.to_string()));
+        }
+    }
+
+    let dir = format!("{}/", request.id);
+    let files = s3_client.extract_files(&dir).await?;
+
+    // Update status to Compiling
+    try_set_compiling_status(db_client, request.id).await?;
+
+    Ok(CompilationInput {
+        workspace_path: Path::new(SOL_ROOT).join(request.id.to_string().as_str()),
+        config: request.config.clone(),
+        contracts: files,
+    })
+}
+pub async fn on_compilation_success(
+    id: Uuid,
+    db_client: &DynamoDBClient,
+    s3_client: &S3Client,
+    compilation_output: CompilationOutput,
+) -> Result<TaskResult, CommandResultHandleError> {
+    const DOWNLOAD_URL_EXPIRATION: Duration = Duration::from_secs(5 * 60 * 60);
```

Comment:
can we move this out from here?

Reply:
The scope of this const is solely the function